### PR TITLE
perf(ingestion/fivetran): Connector performance optimization

### PR DESCRIPTION
**Pull request by shubhamjagtap639**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ class FivetranSource(StatefulIngestionSourceBase):
     platform: str = "fivetran"
 
     def __init__(self, config: FivetranSourceConfig, ctx: PipelineContext):
-        super().__init__(config, ctx)
+        super(FivetranSource, self).__init__(config, ctx)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran.py:69]: why is this necessary?

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py [hunk 2]
```diff
@@ -76,7 +77,7 @@ def _initialize_fivetran_variables(
         )
 
     def _query(self, query: str) -> List[Dict]:
-        logger.debug(f"Query : {query}")
+        logger.debug("Query : {}".format(query))
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py:80]: why was this changed?

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_query.py [hunk 3]
```diff
@@ -31,14 +31,15 @@ def get_user_query(self, user_id: str) -> str:
...
-        return f"""
+        return """
         SELECT connector_id,
         sync_id,
         message_event,
         message_data,
         time_stamp
-        FROM {self.db_clause}log
-        WHERE message_event in ('sync_start', 'sync_end')"""
+        FROM {}log
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_query.py:40]: use named format args

e.g. `"{foo}".format(foo='whatever')`

Comment (shubhamjagtap639) [reply metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_query.py]: Done

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py (hunk 2 above)
Comment (shubhamjagtap639) [reply metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py]: Changes happen automatically. Let me revert it

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/config.py [hunk 4]
```diff
@@ -163,3 +181,7 @@ class FivetranSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigMixin
         default={},
         description="A mapping of destination dataset to platform instance. Use destination id as key.",
     )
+    extract_sync_history_for_interval: int = pydantic.Field(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/fivetran/config.py:184]:
```suggestion
    history_sync_lookback_period: int = pydantic.Field(
```

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/config.py [hunk 5]
```diff
@@ -163,3 +181,7 @@ class FivetranSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigMixin
...
+        7,
+        description="Interval in days to extract connector's sync history from current date.",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/fivetran/config.py:186]:
```suggestion
        description="The number of days to look back when extracting connectors' sync history.",
```

**Review by hsheth2 — COMMENTED** [review]
tweaking variable naming / docs, otherwise looks good

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py [hunk 6]
```diff
@@ -208,50 +214,61 @@ def _get_jobs_list(
             )
         return jobs
 
-    def _get_user_email(self, user_id: Optional[str]) -> Optional[str]:
+    @functools.lru_cache()
+    def get_user_email(self, user_id: str) -> Optional[str]:
         if not user_id:
             return None
         user_details = self._query(
             self.fivetran_log_query.get_user_query(user_id=user_id)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py:222]: is this sending one query per user? ideally this would also do a bulk fetch

File: metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py [hunk 7]
```diff
@@ -272,10 +278,17 @@ def gen_removed_entity_workunits(self) -> Iterable[MetadataWorkUnit]:
...
         # Everything looks good, emit the soft-deletion workunits
         for urn in last_checkpoint_state.get_urns_not_in(
             type="*", other_checkpoint_state=cur_checkpoint_state
         ):
+            if not entity_supports_aspect(guess_entity_type(urn), StatusClass):
+                # If any entity does not support aspect 'status' then skip that entity urn
+                report.report_last_state_non_deletable_entities(urn)
+                continue
             if urn in self._urns_to_skip:
                 logger.debug(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/state/stale_entity_removal_handler.py:293]: makes sense to report these too
```suggestion
                report.report_last_state_non_deletable_entities(urn)
                logger.debug(
```

File: metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py (hunk 6 above)
Comment (shubhamjagtap639) [reply metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py]: Done
